### PR TITLE
Turn channel deletion into a progress tracked, uncancellable, task

### LINKED ISSUE #2140: deleting a channel is a long-running operation but doesn't provide user feedback
* Deleted khan academy (en) channel, 72GB
* From a user's perspective, nothing happened
* Refreshed - still there
* Pressed 'delete' again.

python error:

```
ERROR:sqlalchemy.pool.NullPool:Exception during reset or similar
Traceback (most recent call last):
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/pool.py", line 687, in _finalize_fairy
    fairy._reset(pool)
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/pool.py", line 829, in _reset
    pool._dialect.do_rollback(self)
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 440, in do_rollback
    dbapi_connection.rollback()
ProgrammingError: SQLite objects created in a thread can only be used in that same thread.The object was created in thread id 4672602112 and this is thread id 4681015296
ERROR:sqlalchemy.pool.NullPool:Exception closing connection <sqlite3.Connection object at 0x113b56ce0>
Traceback (most recent call last):
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/pool.py", line 300, in _close_connection
    self._dialect.do_close(connection)
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 446, in do_close
    dbapi_connection.close()
ProgrammingError: SQLite objects created in a thread can only be used in that same thread.The object was created in thread id 4672602112 and this is thread id 4681015296
ERROR:sqlalchemy.pool.NullPool:Exception during reset or similar
Traceback (most recent call last):
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/pool.py", line 687, in _finalize_fairy
    fairy._reset(pool)
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/pool.py", line 829, in _reset
    pool._dialect.do_rollback(self)
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 440, in do_rollback
    dbapi_connection.rollback()
ProgrammingError: SQLite objects created in a thread can only be used in that same thread.The object was created in thread id 4672602112 and this is thread id 4681015296
ERROR:sqlalchemy.pool.NullPool:Exception closing connection <sqlite3.Connection object at 0x111cd65a0>
Traceback (most recent call last):
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/pool.py", line 300, in _close_connection
    self._dialect.do_close(connection)
  File "/Users/d/PythonEnvs/le/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 446, in do_close
    dbapi_connection.close()
ProgrammingError: SQLite objects created in a thread can only be used in that same thread.The object was created in thread id 4672602112 and this is thread id 468101529
```

* CPU was very high for a while
* refreshed again, channel is gone now

----

We should provide some user feedback while the operation is occurring and prevent it from being triggered a second time

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Previously channel deletion was done as a single call to a REST endpoint using DELETE.

This was fine before we were deleting multiple objects in the database and residual unused files, now it can be very slow.

To address this, channel deletion is implemented as a management command, and then called using the tasks API, with progress tracking.

In addition, due to the fact that we can't undo the operation, we do not give the user the option to cancel.

## Issues addressed

Fixes #2140

## Screenshots

Tried to get a screenshot, but it deleted too fast on the tiny channel I tested on.

### CODE REVIEW

**Comment by rtibbles** [comment]
Note this should also fix #2128, by sidestepping the issue.

File: kolibri/plugins/management/assets/src/device_management/views/manage-content-page/task-progress.vue [hunk 1]
```diff
@@ -74,6 +75,13 @@
         required: true,
       },
       id: RequiredString,
+      cancellable: {
+        type: Boolean,
+        required: true,
+      },
+    },
+    watch: {
+      taskHasFailed: 'emitFailure',
```
Comment (jonboiser) [inline kolibri/plugins/management/assets/src/device_management/views/manage-content-page/task-progress.vue]: I would just inline `emitFailure` here, if it's not used by anything else in the component.

File: kolibri/tasks/api.py [hunk 2]
```diff
@@ -185,11 +206,7 @@ def _networkimport(channel_id, update_progress=None, check_for_cancel=None):
             update_progress=update_progress,
             check_for_cancel=check_for_cancel)
     except UserCancelledError:
-        try:
-            os.remove(get_content_database_file_path(channel_id))
-        except OSError:
-            pass
-        # TODO: Delete channel data
+        call_command("deletechannel", channel_id, update_progress=update_progress)
```
Comment (jonboiser) [inline kolibri/tasks/api.py:209]: Could the string `"deletechannel"` here and line 219 be replaced with the `DELETE_CHANNEL` constant above?

Comment (rtibbles) [reply kolibri/tasks/api.py:209]: In this case, no - it's the name of the django management command.

Might be nice to clean this up in the future, by using explicit imports for our `call_command` invocations. https://docs.djangoproject.com/en/1.11/ref/django-admin/#running-management-commands-from-your-code

File: kolibri/plugins/management/assets/src/device_management/views/manage-content-page/task-progress.vue [hunk 3]
```diff
@@ -74,6 +75,17 @@
         required: true,
       },
       id: RequiredString,
+      cancellable: {
```
Comment (jonboiser) [inline kolibri/plugins/management/assets/src/device_management/views/manage-content-page/task-progress.vue:78]: This needs to be defaulted to `true` for the import/export case (throws out prop validation warnings).

Comment (jonboiser) [reply kolibri/plugins/management/assets/src/device_management/views/manage-content-page/task-progress.vue:78]: Nevermind, I think the issue is that this API-to-state mapper actually omits `cancellable`

```
function transformTasks(tasks) {
  return tasks.map(task => ({
    id: task.id,
    type: task.type,
    status: task.status,
    metadata: task.metadata,
    percentage: task.percentage,
  }));
}

```

http://github.com/rtibbles/kolibri/blob/d7d88f3b3cec079e6772845c68c55fbb2e989f1a/kolibri/plugins/management/assets/src/device_management/state/actions/taskActions.js#L13

**Comment by jonboiser** [comment]
## Deleting stuff
Manually tested the deletion and it works well.

## Adding stuff, after deleting stuff

This does seem to fix the symptoms of #2030 (e.g. it adds the channels to the local `pageState` after deleting channels at least once).

Oddly, though, after the import the payload for `SET_CORE_CHANNEL_LIST`, which mutates `core.channels` is still `[]` (so #2128 is possibly still in play). I don't think there are huge consequences of this, except that the 'import preview modal' relies on this list to check whether a channel is already installed (so it no longer has the "already installed" message). Could possibly fix by having that logic rely on `pageState.channelList` instead.

File: kolibri/plugins/management/assets/src/device_management/views/manage-content-page/task-progress.vue (hunk 3 above)
Comment (rtibbles) [reply kolibri/plugins/management/assets/src/device_management/views/manage-content-page/task-progress.vue:78]: Ahh, yes, should change that.

**Comment by rtibbles** [comment]
Yeah, I think removing the reliance of the import app on the `core.channels` is desirable. Can clean up here.

**Comment by rtibbles** [comment]
Have addressed comments, should be good to merge.

**Pull request merged**